gateway-ha/src/main/java/io/trino/gateway/ha/router/BackendStateMBeanExporter.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.gateway.ha.router;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.errorprone.annotations.concurrent.GuardedBy;
+import com.google.inject.Inject;
+import io.trino.gateway.ha.clustermonitor.ClusterStats;
+import io.trino.gateway.ha.clustermonitor.TrinoStatus;
+import jakarta.annotation.PostConstruct;
+import jakarta.annotation.PreDestroy;
+import org.weakref.jmx.MBeanExport;
+import org.weakref.jmx.MBeanExporter;
+import org.weakref.jmx.Managed;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Objects.requireNonNull;
+
+public class BackendStateMBeanExporter
+{
+    @GuardedBy("this")
+    private final List<MBeanExport> mbeanExports = new ArrayList<>();
+
+    private final MBeanExporter exporter;
+    private final BackendStateManager backendStateManager;
+    private final Map<String, ClusterStatsJMX> backendStates = new HashMap<>();
+
+    @Inject
+    public BackendStateMBeanExporter(MBeanExporter exporter, BackendStateManager backendStateManager)
+    {
+        this.exporter = requireNonNull(exporter, "exporter is null");
+        this.backendStateManager = requireNonNull(backendStateManager, "backendStateManager is null");
+    }
+
+    @PostConstruct
+    public synchronized void updateExport()
```

Comment:
Is it possible to make `updateExport()` takes `List<ClusterStats>` from `ClusterStatsObserver`? In that case we don't have to inject BackendStateManager and don't need to add `getAllBackendStates()`.